src_features/signMessageEIP712/ui_logic.c

```diff
@@ -0,0 +1,548 @@
+#ifdef HAVE_EIP712_FULL_SUPPORT
+
+#include <stdlib.h>
+#include <stdbool.h>
+#include "ui_logic.h"
+#include "mem.h"
+#include "mem_utils.h"
+#include "os_io.h"
+#include "shared_context.h"
+#include "ethUtils.h"  // getEthDisplayableAddress
+#include "utils.h"     // uint256_to_decimal
+#include "common_712.h"
+#include "context_712.h"     // eip712_context_deinit
+#include "uint256.h"         // tostring256 && tostring256_signed
+#include "path.h"            // path_get_root_type
+#include "apdu_constants.h"  // APDU response codes
+#include "typed_data.h"
+#include "commands_712.h"
+#include "common_ui.h"
+
+static t_ui_context *ui_ctx = NULL;
```

Comment:
flag globals as such

Reply:
Good idea to rename it so it is more explicit that it is global. :+1: 
Will do in a future PR.